### DISCUSSION TITLE
SVGs for decorators - look pixelated and don't zoom well

### DISCUSSION THREAD

**Discussion by jmenzies12** [eclipse-sirius/sirius-desktop#393] [Q&A]
Hey there,

We want to swap over from using PNG's to SVG's files for our decorators(16x16 pixels), I see that this is supported in Sirius. However I'm getting some odd results, when rendered on the canvas they look quite pixelated and they don't keep their resolution when zooming in. Some of the more rounded icons also end up with jagged edges. When I open the SVG files themselves they look fine in image editors and the browser.

Is this maybe an issue with the default rendering settings used? Or do I need to specify something particular in the SVG.

Example image below shows decorators at 200% zoom.

![image](https://github.com/eclipse-sirius/sirius-desktop/assets/3521863/a08abedf-6e4a-4852-a0d5-14f6468ede62)

Cheers
Jonathan

**Comment by pcdavid** [comment]
I think this is because the decorators rendering does not use the same code as SVG-based nodes, and the default behavior is to render the SVG file (using Batik) at some default resolution and then use the result as a plain bitmap (so pixelated when zoomed).
